monitoring/templates/alert.rules

```diff
@@ -0,0 +1,66 @@
+groups:
+- name: basic
+  rules:
+
+  # Alert for any instance that is unreachable for >5 minutes.
+  - alert: service_down
+    expr: 100  * (count(up == 0) BY (job) / count(up) BY (job)) > 10
+    for: 2m
+    labels:
+      severity: page
+    annotations:
+      summary: "Instance {{ $labels.instance }} down"
+      description: "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 2 minutes."
+
+  - alert: high_load
+    expr: node_load1 > 2
+    for: 5m
+    labels:
+      severity: page
+    annotations:
+      summary: "Instance {{ $labels.instance }} under high load"
+      description: "{{ $labels.instance }} of job {{ $labels.job }} is under high load."
+
+  - alert: DeadMansSwitch
+    expr: vector(1)
+    labels:
+      severity: none
+    annotations:
+      description: This is a DeadMansSwitch meant to ensure that the entire Alerting pipeline
+        is functional.
+      summary: Alerting DeadMansSwitch
+  - alert: NodeExporterDown
+    expr: absent(up{job="node-exporter"} == 1)
+    for: 10m
+    labels:
+      severity: warning
+    annotations:
+      description: Prometheus could not scrape a node-exporter for more than 10m, or node-exporters
+        have disappeared from discovery
+      summary: Prometheus could not scrape a node-exporter
+  - alert: RootDiskRunningFull
+    expr: predict_linear(node_filesystem_free_bytes{mountpoint="/"}[30m], 3600*2) <0
+    for: 10m
+    labels:
+      severity: critical
+    annotations:
+      description: device {{$labels.device}} on node {{$labels.instance}} is running full
+        within the next 2 hours (mounted at {{$labels.mountpoint}})
+      summary: Node disk is running full
+  - alert: MemoryUsageHigh
+    expr: ((node_memory_MemTotal - node_memory_MemFree - node_memory_Buffers - node_memory_Cached) / node_memory_MemTotal) * 100 >= 90
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: '{{$labels.instance}} is using more than 90% Memory for >1h '
+      summary: Instance {{$labels.instance}} Memory usage high
+  - alert: CPUUsageHigh
+    expr: 100 * (1 - avg by(instance) (irate(node_cpu_seconds_total{mode="idle"}[1h]))) >= 90
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: '{{$labels.instance}} is using more than 90% CPU for >1h '
+      summary: Instance {{$labels.instance}} CPU usage high
+  ${custom_alert_rules}
```

Comment:
Not sure about this, but I think that if you call the `indent` function in tf, you'll need to remove all the indentation here.

Reply:
no. This has already been tested